crates/typst-layout/src/math/fragment.rs

```diff
@@ -382,19 +373,22 @@ impl GlyphFragment {
             shift: styles.resolve(TextElem::baseline),
             modifiers: FrameModifiers::get_in(styles),
         };
-        fragment.update_glyph();
-        Ok(Some(fragment))
+        fragment.update_glyph(true);
+        Some(fragment)
     }
 
     /// Sets element id and boxes in appropriate way without changing other
     /// styles. This is used to replace the glyph with a stretch variant.
-    pub fn update_glyph(&mut self) {
+    pub fn update_glyph(&mut self, initial: bool) {
         let id = GlyphId(self.item.glyphs[0].id);
 
         let extended_shape = is_extended_shape(&self.item.font, id);
         let italics = italics_correction(&self.item.font, id).unwrap_or_default();
         let width = self.item.width();
-        if !extended_shape {
+        // The second condition below is needed so we don't accidently add the
+        // italic correction again to the original glyphs. This can happen when
+        // `reset_glyph` is called after the glyph fragment is created.
+        if !extended_shape && (initial || id.0 != self.item.original_id()) {
```

Comment:
This condition here seems slightly hacky. I wonder whether there isn't a cleaner, less mutable way to produce the metrics. But we can also leave that as future work.